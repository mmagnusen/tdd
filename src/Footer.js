import React, { Component } from 'react';
import './App.css';

class Footer extends Component {
    render() {
        return (
            <div>
                <h1>Footer here</h1>
            </div>
        );
    }
}

export default Footer;
