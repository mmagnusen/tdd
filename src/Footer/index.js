import React, { Component } from 'react';

class Footer extends Component {
    render() {
        return (
            <footer>
                <h1>Footer here</h1>
            </footer>
        );
    }
}

export default Footer;
