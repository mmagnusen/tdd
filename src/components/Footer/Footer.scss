
$module: 'Footer';

.#{$module} { 
    padding: 1rem 3rem;
}
  