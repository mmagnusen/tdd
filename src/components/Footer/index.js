import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from '../';
import { LINK_TYPE } from '../../constants';
import navigationActionGenerators from './../../redux/actions/navigationActionGenerators';
import './Footer.scss';

class Footer extends Component {
    updateActiveIndex  = (item, activeIndex) => {
        this.props.dispatch(navigationActionGenerators.updateNavigation({item, activeIndex}))
        window.scrollTo(0, 0)
    }

    render() {
        return (
        <footer className='Footer'>
            <div className='container'>
                <ul>
                    <h4>Jest</h4>
                    <li onClick={() => this.updateActiveIndex('jest', 0)}><Link to='/jest' type={LINK_TYPE.INTERNAL}>What is Jest?</Link></li>
                    <li onClick={() => this.updateActiveIndex('jest', 1)}><Link to='/jest' type={LINK_TYPE.INTERNAL}>Snapshot tests</Link></li>
                    <li onClick={() => this.updateActiveIndex('jest', 2)}><Link to='/jest' type={LINK_TYPE.INTERNAL}>Writing tests</Link></li>
                </ul>
                <ul>
                    <h4>Resources</h4>
                    <li><Link to='/resources/#GeneralTesting' type={LINK_TYPE.ANCHOR}>General testing</Link></li>
                    <li><Link to='/resources/#ForBeginners' type={LINK_TYPE.ANCHOR}>Helpful for beginners</Link></li>
                    <li><Link to='/resources/#Jest' type={LINK_TYPE.ANCHOR}>Jest</Link></li>
                    <li><Link to='/resources/#Selenium' type={LINK_TYPE.ANCHOR}>Selenium</Link></li>
                    <li><Link to='/resources/#GitRepos' type={LINK_TYPE.ANCHOR}>Git repositories</Link></li>
                </ul>
                <ul>
                    <h4>How To Guides</h4>
                    <li onClick={() => this.updateActiveIndex('howTo', 0)}><Link to='/how-to' type={LINK_TYPE.INTERNAL}>Add JavaScript to project</Link></li>
                    <li onClick={() => this.updateActiveIndex('howTo', 1)}><Link to='/how-to' type={LINK_TYPE.INTERNAL}>Add Sass to project</Link></li>
                    <li onClick={() => this.updateActiveIndex('howTo', 2)}><Link to='/how-to' type={LINK_TYPE.INTERNAL}>Run local server</Link></li>
                </ul>
                <ul>
                    <h4>Concepts</h4>
                    <li onClick={() => this.updateActiveIndex('concepts', 0)}><Link to='/concepts' type={LINK_TYPE.INTERNAL}>Scope</Link></li>
                </ul>
            </div>
        </footer>
    )}
};

const mapStateToProps = (state) => {
    return {
        activeIndex: state.navigation.jest
    }
};

export default connect(mapStateToProps)(Footer);
