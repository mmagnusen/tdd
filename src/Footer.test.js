import React from 'react'
import Footer from './Footer'
import { shallow } from 'enzyme';

describe('Footer component', () => {
    it('Renders the Footer with Enzyme', () => {
        const footer = shallow(<Footer />);
        expect(footer).toMatchSnapshot();
    })
})