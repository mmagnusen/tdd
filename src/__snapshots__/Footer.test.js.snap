// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Renders the Footer with Enzyme 1`] = `ShallowWrapper {}`;

exports[`Navigation component Renders the navigation with Enzyme 1`] = `ShallowWrapper {}`;
